archive/archive_ingest/src/test_archive_ingest.py

```diff
@@ -0,0 +1,48 @@
+# -*- encoding: utf-8 -*-
+
+import os
+import archive_ingest
+import pytest
+
+
+def test_sends_location_to_sns(sns_client, topic_arn):
+    os.environ['INGEST_TOPIC_ARN'] = topic_arn
+    request = {'bagURL': 's3://wellcomecollection-assets-archive-ingest/test-bag.zip'}
+
+    response = archive_ingest.handler(event=request, sns_client=sns_client)
+
+    assert response == {'received': 'OK'}
+
+    messages = sns_client.list_messages()
+    assert len(messages) == 1
+    assert messages[0][':message'] == {
+        'namespace': 'wellcomecollection-assets-archive-ingest',
+        'key': 'test-bag.zip'
+    }
+
+
+def test_invalid_url_fails(sns_client, topic_arn):
+    with pytest.raises(ValueError, match="Unrecognised url scheme: invalid"):
+        os.environ['INGEST_TOPIC_ARN'] = topic_arn
+        request = {'bagURL': 'invalid'}
+
+        archive_ingest.handler(event=request, sns_client=sns_client)
```

Comment:
minor: I'd move the first two lines out of the `with` block – leave as little code as possible to fail inside the pytest capture.

Nice find with the `match` parameter, I haven't seen that before.